docs/data/material/components/autocomplete/InfiniteScrolling.js

```diff
@@ -0,0 +1,159 @@
+import * as React from 'react';
+import TextField from '@mui/material/TextField';
+import Autocomplete from '@mui/material/Autocomplete';
+
+export default function InfiniteScrolling() {
+  const [films, setFilms] = React.useState(allFilms.slice(0, 10));
+  const [isLoading, setIsLoading] = React.useState(false);
+
+  const handleOnScrollToBottom = () => {
+    setIsLoading(true);
+    setTimeout(() => {
+      setIsLoading(false);
+      setFilms((previousFilms) => [
+        ...previousFilms,
+        ...allFilms.slice(previousFilms.length, previousFilms.length + 10),
+      ]);
+    }, 2000);
```

Comment:
Same as Joy UI's demo above which is used to prevent showing the flickering `Loading` text when the last film has been reached. 
```suggestion
    if (allFilms.length !== films.length) {
      setIsLoading(true);
      setTimeout(() => {
        setIsLoading(false);
        setFilms((previousFilms) => [
          ...previousFilms,
          ...allFilms.slice(previousFilms.length, previousFilms.length + 10),
        ]);
      }, 2000);
    }
```